lib/StripeClient.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Stripe;
+
+class StripeClient implements StripeClientInterface
+{
+    /**
+     * @var string Default base URL for Stripe's API.
+     */
+    const DEFAULT_API_BASE = 'https://api.stripe.com';
+
+    /**
+     * @var string Default base URL for Stripe's OAuth API.
+     */
+    const DEFAULT_CONNECT_BASE = 'https://connect.stripe.com';
+
+    /**
+     * @var string Default base URL for Stripe's Files API.
+     */
+    const DEFAULT_FILES_BASE = 'https://files.stripe.com';
+
+    private $apiKey;
+    private $clientId;
+
+    private $apiBase;
+    private $connectBase;
+    private $filesBase;
+
+    /**
+     * Initializes a new instance of the {@link StripeClient} class.
+     *
+     * @param string|null $apiKey The API key used by the client to make requests.
+     * @param string|null $clientId The client ID used by the client in OAuth requests.
+     * @param string|null $apiBase The base URL for Stripe's API. Defaults to {@link DEFAULT_API_BASE}.
+     * @param string|null $connectBase The base URL for Stripe's OAuth API. Defaults to {@link DEFAULT_CONNECT_BASE}.
+     * @param string|null $filesBase The base URL for Stripe's Files API. Defaults to {@link DEFAULT_FILES_BASE}.
+     */
+    public function __construct(
+        $apiKey,
```

Comment:
should be nullable

Reply:
To be clear, the argument _is_ nullable, we're just not providing a default value. On second thought, I think this is preferable: it'll likely be very uncommon to create a client without an API key, so I think it's more helpful to require the argument. Users who don't want to set an API key on the client will just have to pass `null` explicitly.